script/deploy/tron/register-facets-to-diamond.ts

```diff
@@ -0,0 +1,750 @@
+  ['DexManagerFacet', 'PeripheryRegistryFacet'], // Configuration
+  ['GenericSwapFacet', 'GenericSwapFacetV3'], // Swap functionality
+  ['CalldataVerificationFacet', 'EmergencyPauseFacet'], // Security
+]
+
+/**
+ * Estimate energy for diamondCut transaction using triggerconstantcontract
+ */
+async function estimateDiamondCutEnergy(
+  tronWeb: any,
+  diamondAddress: string,
+  facetCuts: any[],
+  fullHost: string
+): Promise<number> {
+  try {
+    consola.info(
+      ' Calling triggerconstantcontract API for energy estimation...'
+    )
+
+    // Diamond address should stay in base58 for Tron API
+
+    // Encode parameters (without function selector)
+    // facetCuts is already formatted as arrays
+    const encodedParams = tronWeb.utils.abi.encodeParams(
+      ['(address,uint8,bytes4[])[]', 'address', 'bytes'],
+      [facetCuts, '0x0000000000000000000000000000000000000000', '0x']
+    )
+
+    // Function selector for diamondCut
+    const functionSelector =
+      'diamondCut((address,uint8,bytes4[])[],address,bytes)'
+
+    // Make API call to triggerconstantcontract
+    const apiUrl =
+      fullHost.replace(/\/$/, '') + '/wallet/triggerconstantcontract'
+
+    const payload = {
+      owner_address: tronWeb.defaultAddress.base58,
+      contract_address: diamondAddress, // Use base58 format for Tron API
+      function_selector: functionSelector,
+      parameter: encodedParams.replace('0x', ''),
+      fee_limit: 1000000000, // High limit for estimation only
+      call_value: 0,
+      visible: true,
+    }
+
+    consola.info('📤 Sending payload to:', apiUrl)
+    consola.info('Payload:', {
+      owner_address: payload.owner_address,
+      contract_address: payload.contract_address,
+      function_selector: payload.function_selector,
+      parameter_length: payload.parameter.length,
+      parameter_preview: payload.parameter.substring(0, 100) + '...',
+      fee_limit: payload.fee_limit,
+      call_value: payload.call_value,
+      visible: payload.visible,
+    })
+
+    const response = await fetch(apiUrl, {
+      method: 'POST',
+      headers: {
+        accept: 'application/json',
+        'content-type': 'application/json',
+      },
+      body: JSON.stringify(payload),
+    })
+
+    if (!response.ok) {
+      const errorText = await response.text()
+      throw new Error(`API call failed: ${response.status} - ${errorText}`)
+    }
+
+    const result = await response.json()
+
+    consola.info(' Estimation API response:', JSON.stringify(result, null, 2))
+
+    if (result.result?.result === false)
+      throw new Error(
+        `Energy estimation failed: ${
+          result.result?.message || JSON.stringify(result)
+        }`
+      )
+
+    if (result.energy_used) {
+      consola.info(` Raw energy estimate: ${result.energy_used}`)
+      // The actual transaction uses much more energy than the estimate
+      // Multiply by 10x for safety (actual usage was ~16x the estimate)
```

Comment:
If actual usage was 16x, why do you use 10 as multiplier here?

Reply:
Fixed: ddd19d07